server/infinispan/src/main/java/org/hawkular/apm/server/infinispan/InfinispanSpanCache.java

```diff
@@ -120,6 +121,16 @@ public void store(String tenantId, List<Span> spans, Function<Span, String> cach
             }
 
             spansCache.put(cacheKeyEntrySupplier.apply(span), span, 1, TimeUnit.MINUTES);
+
+            if (span.getParentId() != null && !span.serverSpan()) {
+                Set<Span> children = childrenCache.get(span.getParentId());
+                if (children == null) {
+                    children = new HashSet<>();
+                    childrenCache.put(span.getParentId(), children);
+                }
```

Comment:
Need to do 'put' after the set has been changed. In case the put results in the value being written to disk or distributed to clustered caches.